nima/webserver/static-content/src/main/java/io/helidon/nima/webserver/staticcontent/CachedHandlerJar.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package io.helidon.nima.webserver.staticcontent;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.function.BiConsumer;
+
+import io.helidon.common.configurable.LruCache;
+import io.helidon.common.http.Http;
+import io.helidon.common.http.ServerResponseHeaders;
+import io.helidon.common.media.type.MediaType;
+import io.helidon.nima.webserver.http.ServerRequest;
+import io.helidon.nima.webserver.http.ServerResponse;
+
+import static io.helidon.nima.webserver.staticcontent.FileBasedContentHandler.contentLength;
+import static io.helidon.nima.webserver.staticcontent.FileBasedContentHandler.send;
+import static io.helidon.nima.webserver.staticcontent.StaticContentHandler.processEtag;
+import static io.helidon.nima.webserver.staticcontent.StaticContentHandler.processModifyHeaders;
+
+record CachedHandlerJar(Path path,
+                        MediaType mediaType,
+                        Instant lastModified,
+                        BiConsumer<ServerResponseHeaders, Instant> setLastModifiedHeader) implements CachedHandler {
+    private static final System.Logger LOGGER = System.getLogger(CachedHandlerJar.class.getName());
+
+    @Override
+    public boolean handle(LruCache<String, CachedHandler> cache,
+                          Http.Method method,
+                          ServerRequest request,
+                          ServerResponse response,
+                          String requestedResource) throws IOException {
+
+        // check if file still exists (the tmp may have been removed, file may have been removed
+        // there is still a race change, but we do not want to keep cached records for invalid files
+        if (!Files.exists(path)) {
+            cache.remove(requestedResource);
+            return false;
+        }
```

Comment:
would file watcher help here to offload the check from handler?

Reply:
The check needs to touch the file system, nevertheless the only case the file may get removed is when somebody (or the OS) cleans up the temporary directory we use to extract files. I think file watcher is not a good match here